addons/download-button/userscript.js

```diff
@@ -0,0 +1,57 @@
+import downloadBlob from "../../libraries/common/cs/download-blob.js";
+
+export default async function ({ addon, console, msg }) {
+  const vm = addon.tab.traps.vm;
+  const { redux } = addon.tab;
+
+  await redux.waitForState((state) => typeof state.session.session?.user === "object");
+  const username = await addon.auth.fetchUsername();
```

Comment:
Why use both of these methods? The first just cases the addon to never do anything when logged out.

Reply:
genuinely no idea, obviously we want them to still be able to download other's projects when they are logged out... ill deal with it